src/Tmds.Ssh/PrivateKeyParser.OpenSsh.cs

```diff
@@ -0,0 +1,279 @@
+        {
+            privateKeyList = reader.ReadStringAsBytes();
+        }
+        else if (passphrase.Length == 0)
+        {
+            error = new FormatException("Key was encrypted but no passphrase was provided.");
+            return false;
+        }
+        else
+        {
+            if (!TryDecryptOpenSshPrivateKey(reader, cipherName, kdfName, kdfOptions, passphrase, out var decryptedKey, out error))
+            {
+                return false;
+            }
+            privateKeyList = new ReadOnlySequence<byte>(decryptedKey);
+        }
+
+        reader = new SequenceReader(privateKeyList);
+        /*
+            uint32	checkint
+            uint32	checkint
+            byte[]	privatekey1
+            string	comment1
+            byte[]	privatekey2
+            string	comment2
+            ...
+            byte[]	privatekeyN
+            string	commentN
+            byte	1
+            byte	2
+            byte	3
+            ...
+            byte	padlen % 255
+        */
+        uint checkInt1 = reader.ReadUInt32();
+        uint checkint2 = reader.ReadUInt32();
+        if (checkInt1 != checkint2)
+        {
+            error = new FormatException($"The checkints mismatch. The key is invalid or the passphrase is wrong.");
+            return false;
+        }
+
+        Name keyType = reader.ReadName();
+        if (keyType == AlgorithmNames.SshRsa)
+        {
+            return TryParseOpenSshRsaKey(reader, out privateKey, out error);
+        }
+        if (keyType.ToString().StartsWith("ecdsa-sha2-"))
+        {
+            return TryParseOpenSshEcdsaKey(keyType, reader, out privateKey, out error);
+        }
+        else
+        {
+            error = new NotSupportedException($"The key type is unsupported: '{keyType}'.");
+            return false;
+        }
+    }
+
+    private static bool TryDecryptOpenSshPrivateKey(
+        SequenceReader reader,
+        Name cipher,
+        Name kdf,
+        ReadOnlySequence<byte> kdfOptions,
+        ReadOnlySpan<byte> passphrase,
+        [NotNullWhen(true)] out byte[]? privateKey,
+        [NotNullWhen(false)] out Exception? error)
+    {
+        privateKey = null;
+
+        if (kdf != AlgorithmNames.Bcrypt)
+        {
+            error = new NotSupportedException($"Unsupported KDF: '{kdf}'.");
+            return false;
+        }
+
+        /*
+            string salt
+            uint32 rounds
+        */
+        var kdfReader = new SequenceReader(kdfOptions);
+        ReadOnlySequence<byte> kdfSalt = kdfReader.ReadStringAsBytes();
+        uint rounds = kdfReader.ReadUInt32();
+
+        if (!OpenSshKeyCipher.TryGetCipher(cipher, out var keyCipher))
+        {
+            error = new NotSupportedException($"Unsupported Cipher: '{cipher}'.");
+            return false;
+        }
+
+        try
+        {
+            byte[] derivedKey = new byte[keyCipher.KeyLength + keyCipher.IVLength];
+            new BCrypt().Pbkdf(
```

Comment:
Can you add a static method to `BCrypt` that we call here, and name it something like:
`DeriveKeyFromPassword`. The static method can call what you are calling here.

Reply:
I made this a static method so kept the `Pbkdf` name, let me know if you prefer `DeriveKeyFromPassword`.

Reply:
Yes, I prefer a more descriptive name for the method.

Reply:
No worries, have updated the code.